README.md

```diff
@@ -1,11 +1,99 @@
 # Topiary
 
-![CI Status](https://github.com/tweag/topiary/actions/workflows/ci.yml/badge.svg)
+![CI Status][ci-badge]
 
 Topiary aims to be a uniform formatter for simple languages, as part of
-the [Tree-sitter](https://tree-sitter.github.io/tree-sitter/) ecosystem.
-It is named after the art of clipping or trimming trees into fantastic
-shapes.
+the [Tree-sitter] ecosystem. It is named after the art of clipping or
+trimming trees into fantastic shapes.
+
+Topiary is designed for formatter authors and formatter users. Authors
+can create a formatter for a language without having to write their own
+parser. Users benefit from uniform code style and, potentially, the
+convenience of using a single formatter tool across multiple languages
+over their codebases.
+
+## Motivation
+
+The style in which code is written has, historically, been mostly left
+to personal choice. Of course, this is subjective by definition and has
+led to many wasted hours reviewing formatting choices, rather than the
+code itself. Prescribed style guides were an early solution to this,
+spawning tools that lint a developer's formatting and ultimately leading
+to automatic formatters. The latter were popularised by
+[`gofmt`][gofmt], whose developers had [the insight][gofmt-slides] that
+"good enough" uniform formatting, imposed on a codebase, with no scope
+for subjective adjustment, largely resolves these problems.
+
+Topiary follows this trend by aspiring to be a "universal formatter
+engine", which allows developers to not only automatically format their
+codebases with a uniform style, but to define that style for new
+languages using a [simple DSL][tree-sitter-query]. This allows for the
+rapid prototyping of formatters, providing a [Tree-sitter
+grammar][tree-sitter-parsers] is defined for that language.
+
+## Design Principles
+
+Topiary has been created with the following goals in mind:
+
+* Use [Tree-sitter] for parsing, to avoid writing yet another grammar
+  for a formatter.
+
+* For bundled formatting styles -- for a handful of languages where no
+  de facto formatter exists -- to meet the following constraints:
+
+  * Be compatible with attested formatting styles used for that language
+    in the wild.
+
+  * Be faithful to the author's intent: If code has been written such
+    that it spans multiple lines, that decision is preserved.
+
+  * Minimise changes between commits such that diffs focus mainly on the
+    code that's changed, rather that superficial artefacts.
```

Comment:
I'm not sure that this is a good description of diff minimisation. The diff minimisation goal is to make a style where 1/ change on one line doesn't format others (I think that's what you are describing) 2/ (and, I'd say, it is more often where arbitrage is) a formatting style that doesn't require you to modify a line spuriously.

(2) is the reason for normalising terminators when we can (we currently can't in Ocaml, but this is why this style is desired):
```ocaml
[
  1;
  2;
]
```

Reply:
Expanded in cd20b49a278a